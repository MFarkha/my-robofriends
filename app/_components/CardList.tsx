import React from "react";
import Card from "./Card";
import { IRobot } from "../page";

type CardListProps = {
  error: string;
  robots: Array<IRobot>;
};

const CardList = ({ error, robots }: CardListProps) => {
  if (error.length !== 0) {
    throw new Error();
  }
  return (
    <div>
      {robots.map((robot) => {
        return <Card robot={robot} key={robot.id} />;
      })}
    </div>
  );
};

export default CardList;
