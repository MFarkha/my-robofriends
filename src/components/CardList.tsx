import Card from './Card';
import { IRobot } from '../app/store';

type CardListProps = {
    robots: Array<IRobot>
};

const CardList = ({ robots }: CardListProps) => {
    return (
        <div>
            {
                robots.map((robot) => {
                    return (
                        <Card robot={robot} key={robot.id}/>
                    );
                })
            }
        </div>
    );
}

export default CardList;