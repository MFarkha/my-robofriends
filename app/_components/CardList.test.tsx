import React from "react";
import { expect, test } from "vitest";
import { render } from "@testing-library/react";
import CardList from "./CardList";

test("expects to render the CardList component", () => {
  const mockRobots = [
    {
      key: 10,
      id: 1,
      name: "John Snow",
      email: "john@example.com",
    },
    {
      key: 11,
      id: 2,
      name: "Kelly Gale",
      email: "kelly@example.com",
    },
  ];
  const { container } = render(<CardList error="" robots={mockRobots} />);
  expect(container).toMatchSnapshot();
});
