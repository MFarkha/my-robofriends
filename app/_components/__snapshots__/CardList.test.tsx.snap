// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`expects to render the CardList component 1`] = `
<div>
  <div>
    <div
      class="bg-light-green dib br3 pa3 ma2 grow bw2 shadow-5"
    >
      <img
        alt="robots"
        data-nimg="1"
        decoding="async"
        height="200"
        loading="lazy"
        src="/_next/image?url=https%3A%2F%2Frobohash.org%2F1.png%3Fsize%3D200x200&w=640&q=75"
        srcset="/_next/image?url=https%3A%2F%2Frobohash.org%2F1.png%3Fsize%3D200x200&w=256&q=75 1x, /_next/image?url=https%3A%2F%2Frobohash.org%2F1.png%3Fsize%3D200x200&w=640&q=75 2x"
        style="color: transparent;"
        width="200"
      />
      <div>
        <h2
          class="f5"
        >
          John Snow
        </h2>
        <p>
          john@example.com
        </p>
      </div>
    </div>
    <div
      class="bg-light-green dib br3 pa3 ma2 grow bw2 shadow-5"
    >
      <img
        alt="robots"
        data-nimg="1"
        decoding="async"
        height="200"
        loading="lazy"
        src="/_next/image?url=https%3A%2F%2Frobohash.org%2F2.png%3Fsize%3D200x200&w=640&q=75"
        srcset="/_next/image?url=https%3A%2F%2Frobohash.org%2F2.png%3Fsize%3D200x200&w=256&q=75 1x, /_next/image?url=https%3A%2F%2Frobohash.org%2F2.png%3Fsize%3D200x200&w=640&q=75 2x"
        style="color: transparent;"
        width="200"
      />
      <div>
        <h2
          class="f5"
        >
          Kelly Gale
        </h2>
        <p>
          kelly@example.com
        </p>
      </div>
    </div>
  </div>
</div>
`;
